Tests/test_SeqIO_UniprotIO.py

```diff
@@ -24,7 +24,7 @@ def test_uni001(self):
 
         datafile = os.path.join("SwissProt", filename)
 
-        with open(datafile) as handle:
+        with open(datafile, "rb") as handle:
```

Comment:
Do these changes break historic support for handles opened in text mode?

(I agree XML files ought to be opened in binary mode, but am worried about a backward compatibility break)

Reply:
Yes it will break. On the other hand, opening a handle as text may give you a bug. Pick your poison.

Reply:
If you are concerned about this, you could modify the `SequenceIterator` base class and allow `mode="bt"` and  `mode="tb"`in addition to  `mode="b"` and  `mode="t"`.
The option `mode="bt"` would mean that both binary and text are accepted, but binary is preferred.
The option `mode="tb"` would mean that both binary and text are accepted, but text is preferred (I don't know if there is much use for this).

Having an option `mode="bt"` would also be useful for the Fasta parser, as parsing a file as binary is much faster than parsing a file as text, while for everyday usage it is convenient to still allow parsing a file as text.

Reply:
I like that plan.

Reply:
I can add that as a separate PR.

Reply:
OK then - thank you